Synergy/Controllers/RandomCodeController.cs

```diff
@@ -0,0 +1,76 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Synergy.Authorization;
+using Synergy.Models;
+using Synergy.Services;
+using Permissions = Synergy.Authorization.Permissions;
+
+namespace Synergy.Controllers;
+
+[ApiController]
+[Route("/api/[controller]")]
+public class RandomCodeController : ControllerBase
+{
+    private readonly RandomCodeService _randomCodeService;
+
+    public RandomCodeController(RandomCodeService randomCodeService)
+    {
+        _randomCodeService = randomCodeService;
+    }
+
+    [HttpPost("generate-code")]
+    [Authorize]
+    [AuthorizePermission(Permissions.CanRegisterTeam)]
+    public async Task<IActionResult> GenerateCode([FromBody] TeamId body)
+    {
+        var teamId = body.Id;
+        try
+        {
+            var randomCode = await _randomCodeService.GenerateRandomCodeInvite(teamId);
+            return Ok(randomCode);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, "Internal Server error when generating random code");
+        }
+    }
+
+    [HttpPost("validate-invitation-code")]
+    public async Task<IActionResult> ValidateCode([FromBody] RandomCodes body)
```

Comment:
For the types that are used in the bodies here. Usually we want a suffix for the type, so:
`Input` if the type is ONLY used as part of input data for a request. So this would be `RandomCodesInput`
`Response` if the type is ONLY used to return data from a request. 
`Dto` if the request is used for both input and response.